pkg/server/datastore/datastore.go

```diff
@@ -183,6 +187,17 @@ type ListRegistrationEntriesResponse struct {
 	Pagination *Pagination
 }
 
+type ListRegistrationEntriesEventsRequest struct {
+	Pagination         *Pagination
+	GreaterThanEventID uint
+}
+
+type ListRegistrationEntriesEventsResponse struct {
+	EntryIDs     []string
+	FirstEventID uint
```

Comment:
What this FirstEventID is? 
is it id of the first elment in the response? how is it passed (or affected) on pagination?

Reply:
yes first id of element in responses.

Reply:
Im not sure about pagination. let me test

Reply:
Do we need pagination for this api?

Reply:
I'd put it in.   Bulk deletion / addition APIs could (even though the data per event is small) overflow the single page response.

Reply:
I tried to create unit tests for the pagination but it seems to overlap some with the specific [greater than event id filter](https://github.com/spiffe/spire/pull/4411/files/8c608d6ef7c754717438c0fd417f43cfcf57431b#diff-958c4d2c40aabdcceb9a181b5671b180ea122cc24d8d08dcf09498ef8a46bf92R192) for this api. This is similar to the Token half of the pagination, but with an unlimited page size. Thats what we need for this functionality, all events that happened since the last time we polled the api. 

Given the current implementation [fetches all entries with no pagination](https://github.com/spiffe/spire/blob/main/pkg/server/cache/entrycache/fullcache_ds.go#L46-L50), im not sure pagination is needed here. @edwbuck what do you think of removing the pagination. This would simplify the implementation.

Reply:
I don't have an issue with non-pagination.  I just wanted to mention the bulk updates.